dm/pkg/utils/db.go

```diff
@@ -654,3 +664,8 @@ func GetMaxConnectionsForConn(ctx context.Context, conn *sql.Conn) (int, error)
 	maxConnections, err := strconv.ParseUint(maxConnectionsStr, 10, 32)
 	return int(maxConnections), err
 }
+
+// IsMariaDB tells whether the version is mariadb.
+func IsMariaDB(version string) bool {
+	return strings.Contains(strings.ToUpper(version), "MARIADB")
```

Comment:
Here the `IsMariaDB` is moved from dm/pkg/checker/utils.go.

At last time, it is moved from tidb-tools/pkg/check/utils.go https://github.com/pingcap/tiflow/pull/3770.

So I just move it to a more proper place, because `GetFlavor` also call it.